### PR TITLE
Make planetiler-openmaptiles runnable as a standalone project

### PR DESCRIPTION
**Pull request by msbarry**
Set up this project so it's runnable in a standalone context:

- Moved `pom.xml` to `submodule.pom.xml` when used as a submodule, add new `pom.xml` that can be run standalone from this project without embedding as a submodule
- Set up `pom.xml` to produce a runnable docker container and jar file
- Got tests passing
- Setup spotless autoformatting
- Added regenerate-openmpatiles script to generate code from openmaptiles repo (and a github action to verify)
- Setup dependabot
- Moved classes from `com.onthegomap.planetiler.openmaptiles` to `org.openmaptiles`
- Build a docker image on each commit to main and publish to `openmaptiles/planetiler-openmaptiles`
- Update instructions in README

The performance github actions might not pass before we land this since we need the base ref for a PR to have the new structure first (next one should pass). Also submodule might not pass until after we land this and update planetiler repo to point to new `submodule.pom.xml` in https://github.com/onthegomap/planetiler/pull/312

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -94,101 +67,78 @@ jobs:
...
+          repository: onthegomap/planetiler
+          ref: main
           path: planetiler
       - name: Checkout this PR planetiler-openmaptiles repo
         uses: actions/checkout@v3
         with:
           path: planetiler-openmaptiles
+      # TODO remove, added just for testing
+      - name: TODO TEMP FOR TESTING
+        run: cp planetiler-openmaptiles/submodule.pom.xml planetiler-openmaptiles/pom.xml
```
Comment (msbarry) [inline .github/workflows/maven.yml:128]: TODO remove this

File: .github/workflows/performance.yml [hunk 2]
```diff
@@ -26,36 +26,21 @@ jobs:
...
       - name: 'Checkout PT-OMT base'
         uses: actions/checkout@v3
         with:
-          path: base/planetiler-openmaptiles
-          ref: ${{ github.event.pull_request.base.sha }}
-      - name: 'Move planetiler-openmaptiles into planetiler for base'
-        run: mv base/planetiler-openmaptiles base/planetiler
+          path: base
+          # TODO removed for testing
+          # ref: ${{ github.event.pull_request.base.sha }}
```
Comment (msbarry) [inline .github/workflows/performance.yml:40]: TODO add back in

**Comment by msbarry** [comment]
> I created `DOCKERHUB_PASSWORD` and `DOCKERHUB_PASSWORD` secrets for this repo which can be used.

Is one of them `DOCKERHUB_USER` or `DOCKERHUB_USERNAME`?  I added a [snapshot workflow](https://github.com/openmaptiles/planetiler-openmaptiles/pull/19/files#diff-4c087648fbb704b0cc106eebfff7779856a6af4a6987e919debdd3afd1fdf7bb) that _should_ publish a new `latest` image to docker hub on each commit to main.

**Comment by lazaa32** [comment]
> > I created `DOCKERHUB_PASSWORD` and `DOCKERHUB_PASSWORD` secrets for this repo which can be used.
> 
> Is one of them `DOCKERHUB_USER` or `DOCKERHUB_USERNAME`? I added a [snapshot workflow]

Sorry, it's `DOCKERHUB_USERNAME` and `DOCKERHUB_PASSWORD`

**Comment by msbarry** [comment]
OK thanks @lazaa32, updated the PR. It should be ready to merge now - performance and submodule checks won't work until after it lands and I update planetiler (https://github.com/onthegomap/planetiler/pull/312).  Let me know if there's anything else you wanted me to change on this before merging.

**Review by lazaa32 — APPROVED** [review]

**Review by lazaa32 — APPROVED** [review]
Thanks @msbarry I think we are all good to merge.

**Comment by msbarry** [comment]
Thanks @lazaa32 I merged this and it looks like CI from other PRs is passing now (see #20), but it looks like the snapshot publish failed on main.  Can you try running:

```
./mvnw -B -ntp -Pjib-multi-arch \
            -Dimage.tags="latest" \
            -Djib.to.auth.username="docker username" \
            -Djib.to.auth.password="docker password" \
            package jib:build --file pom.xml
```

locally on your machine to see if the publish succeeds?

**Comment by lazaa32** [comment]
I got the same error when trying it locally.
This is how we use it with openmaptiles-tools push workflow, where it works fine:
https://github.com/openmaptiles/openmaptiles-tools/blob/master/.github/workflows/push.yml

**Pull request merged**